src/components/tl/ucp/gatherv/gatherv_linear.c

```diff
@@ -0,0 +1,125 @@
+    ucc_rank_t         grank = UCC_TL_TEAM_RANK(team);
+    ucc_rank_t         gsize = UCC_TL_TEAM_SIZE(team);
+    int                polls = 0;
+    ucc_rank_t         peer, posts, nreqs;
+    size_t             dt_size, data_size, data_displ;
+
+    if (UCC_IS_ROOT(*args, grank)) {
+        posts   = UCC_TL_UCP_TEAM_LIB(team)->cfg.gatherv_linear_num_posts;
+        nreqs   = (posts > gsize || posts == 0) ? gsize : posts;
+        dt_size = ucc_dt_size(TASK_ARGS(task).dst.info_v.datatype);
+
+        while (polls++ < task->n_polls) {
+            while ((task->tagged.recv_posted < gsize) &&
+                   ((task->tagged.recv_posted - task->tagged.recv_completed) <
+                    nreqs)) {
+                peer       = get_peer(grank, gsize, task->tagged.recv_posted);
+                data_size  = ucc_coll_args_get_count(args,
+                                args->dst.info_v.counts, peer) * dt_size;
+                data_displ = ucc_coll_args_get_displacement(args,
+                                args->dst.info_v.displacements, peer) * dt_size;
+                UCPCHECK_GOTO(ucc_tl_ucp_recv_nz(PTR_OFFSET(rbuf, data_displ),
+                                                 data_size, rmem, peer, team, task),
+                               task, out);
+                polls = 0;
+            }
+        }
+        if (task->tagged.recv_posted < gsize) {
+            return;
+        }
+    }
+
+    task->super.status = ucc_tl_ucp_test(task);
+out:
+    if (task->super.status != UCC_INPROGRESS) {
+        UCC_TL_UCP_PROFILE_REQUEST_EVENT(coll_task,
+                                         "ucp_gatherv_pairwise_done", 0);
+    }
+    return;
+}
+
+ucc_status_t ucc_tl_ucp_gatherv_linear_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task  = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+    ucc_tl_ucp_team_t *team  = TASK_TEAM(task);
+    ucc_coll_args_t   *args  = &TASK_ARGS(task);
+    ucc_rank_t         grank = UCC_TL_TEAM_RANK(team);
+    ucc_memory_type_t  smem  = args->src.info.mem_type;
+    ucc_memory_type_t  rmem  = args->dst.info_v.mem_type;
+    void *             sbuf  = args->src.info.buffer;
+    void              *rbuf;
+    size_t             dt_size, data_displ, data_size;
+
+    UCC_TL_UCP_PROFILE_REQUEST_EVENT(coll_task, "ucp_gatherv_pairwise_start",
+                                     0);
+    ucc_tl_ucp_task_reset(task, UCC_INPROGRESS);
+
+    if (UCC_IS_ROOT(*args, grank)) {
+        if (!UCC_IS_INPLACE(*args)) {
+            dt_size    = ucc_dt_size(args->dst.info_v.datatype);
+            data_size  = ucc_coll_args_get_count(args,
+                            args->dst.info_v.counts, grank) * dt_size;
+            data_displ = ucc_coll_args_get_displacement(args,
+                            args->dst.info_v.displacements, grank) * dt_size;
+            rbuf       = PTR_OFFSET(args->dst.info_v.buffer, data_displ);
+
+            UCPCHECK_GOTO(ucc_tl_ucp_recv_nz(rbuf, data_size, rmem, grank, team,
```

Comment:
movesend & recv  to progress instead of start? (similar to scatterv comment in PR #663 )